wlauto/instrumentation/poller/__init__.py

```diff
@@ -0,0 +1,68 @@
+#    Copyright 2015 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=access-member-before-definition,attribute-defined-outside-init,unused-argument
+
+from wlauto import Instrument, Parameter, Executable
+from wlauto.exceptions import ConfigError
+from wlauto.utils.types import list_of_strs
+
+
+class Poller(Instrument)
+    name = 'poller'
+    description = """
+    Polls the given files at a set sample interval. The values are output in CSV format.
+    """
+
+    parameters = [
+        Parameter('sample_interval', kind=int, default=1000,
+                  description="""The interval between samples in mS."""),
+        Parameter('files', kind=list_of_strs,
+                  description="""A list of paths to the files to be polled"""),
+        Parameter('as_root', kind=bool, default=True,
+                  description="""
+                  Whether or not the poller will be run as root. This should be
+                  used when the file you need to poll can only be accessed by root.
+                  """),
+    ]
+
+    def validate(self):
+        if not self.device.is_rooted and self.as_root:
+            raise ConfigError('The device is not rooted, cannot run poller as root.')
+
+    def initialize(self, context):
+        #for file_ in self.files:
+        #    print self.device.execute("if [ -e '{}' ]; then echo 1; else echo 0; fi".format(file_))
+        #    if not self.device.is_file(file_):
+        #        raise ConfigError('The file {} does not exist'.format(file_))
+
+        host_poller = context.resolver.get(Executable(self, self.device.abi,
+                                                      "poller"))
+        target_poller = self.device.install_if_needed(host_poller)
+
+        self.target_output_path = self.device.path.join(self.device.working_directory, 'poller.csv')
+        self.command = '{} {} {} > {}'.format(target_poller,
+                                              self.sample_interval * 1000,
+                                              ' '.join(self.files),
+                                              self.target_output_path)
+
+    def start(self, context):
+        self.device.kick_off(self.command, as_root=self.as_root)
+
+    def stop(self, context):
+        self.device.killall('poller', signal='TERM', as_root=self.as_root)
+
+    def update_result(self, context):
+        self.device.pull_file(self.target_output_path, context.output_directory)
```

Comment:
Should delete `self.target_output_path` inside `teardown()`. Depending on the number of files monitored, polling rate and the duration of the run, these files can potentially be quite sizeable -- we don't want to lave them hanging around.